### DISCUSSION TITLE
Remove index mapping from email text

### DISCUSSION THREAD

**Discussion by ngms17** [jertel/elastalert2#803] [Q&A] [answered]
When sending an email after a query hit, my email shows the text that i input in the rule configuration file, but also shows all the information from the index mapping. How can i remove the index mapping information?

**Comment by jertel** [comment, marked as answer]
See the documentation for [alert formatting](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#alert-subject)

**Reply by ngms17** [reply]
Also, the timestamp associated with the alert is in UTC. I´m in Europe/Lisbon timezone. How can i change that?

**Reply by jertel** [reply]
Possibly with additional options. Some may not be configurable without using an Enhancement.

https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=format%20utc#use-local-time

https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=format%20utc#timestamp-format

**Reply by ngms17** [reply]
Found a possible solution using enhancements:

```
from elastalert.enhancements import BaseEnhancement

class TimeEnhancement(BaseEnhancement):
    def process(self, match):
        for k, v in match.items():
            if isinstance(v, str) and v.endswith('Z'):
                try:
                    match[k] = pretty_ts(v)
                except:
                    pass
```

This works but it fail to parse the filed "match_time":

```
elasticsearch.exceptions.RequestError: RequestError(400, 'mapper_parsing_exception', "failed to parse field [match_time] of type [date] in document with id 'P1Bm-n8BvcijQIY5yRME'. Preview of field's value: '2022-04-05 16:44 WEST'")
WARNING:elasticsearch:POST http://xxxx:9200/elastalert_status/_doc [status:400 request:0.003s]
ERROR:root:Error writing alert info to Elasticsearch: RequestError(400, 'mapper_parsing_exception', "failed to parse field [match_time] of type [date] in document with id '5VBm-n8BvcijQIY5zRs1'. Preview of field's value: '2022-04-05 16:44 WEST'")
```
